router/batchrouter/asyncdestinationmanager/bing-ads/token.go

```diff
@@ -62,8 +84,78 @@ func (ts *tokenSource) generateToken() (*secretStruct, error) {
 	return &secret, nil
 }
 
+func (ts *tokenSource) generateTokenV2() (*secretStruct, error) {
+	destination := oauthv2.DestinationInfo{
+		WorkspaceID:    ts.workspaceID,
+		DefinitionName: ts.destinationName,
+		ID:             ts.destinationID,
+	}
+	refreshTokenParams := oauthv2.RefreshTokenParams{
+		WorkspaceID: ts.workspaceID,
+		DestDefName: ts.destinationName,
+		AccountID:   ts.accountID,
+		Destination: &destination,
+	}
+	statusCode, authResponse, err := ts.oauthClientV2.FetchToken(&refreshTokenParams)
+	if err != nil {
+		if authResponse != nil {
+			return nil, fmt.Errorf("fetching access token: %v, %d", authResponse.Err, statusCode)
+		}
+		return nil, fmt.Errorf("fetching access token resulted in an error: %v,%d", err, statusCode)
+	}
```

Comment:
```suggestion
	if err != nil && authResponse != nil {
		return nil, fmt.Errorf("fetching access token: %v, %d", authResponse.Err, statusCode)
	}
	if err != nil {
		return nil, fmt.Errorf("fetching access token resulted in an error: %v,%d", err, statusCode)
	}
```

Reply:
Also it isn't idiomatic to have a non nil error with a non nil auth response. The reason for the error should be deduced based on the error and not the authResponse

Reply:
This is done.